boards/DCM/Src/App/states/App_InitState.c

```diff
@@ -23,12 +29,35 @@ static void InitStateRunOnTick100Hz(struct StateMachine *const state_machine)
     App_AllStatesRunOnTick100Hz(state_machine);
 
     struct DcmWorld *world = App_SharedStateMachine_GetWorld(state_machine);
+    struct DcmCanRxInterface *can_rx_interface = App_DcmWorld_GetCanRx(world);
     struct DcmCanTxInterface *can_tx_interface = App_DcmWorld_GetCanTx(world);
+    struct ErrorTable *       error_table  = App_DcmWorld_GetErrorTable(world);
+    struct StartSwitch *      start_switch = App_DcmWorld_GetStartSwitch(world);
 
     App_CanTx_SetPeriodicSignal_TORQUE_REQUEST(can_tx_interface, 0.0f);
 
-    // No need for any safety checks, just run! (this is a demo)
-    App_SharedStateMachine_SetNextState(state_machine, App_GetDriveState());
+    bool any_critical_errors =
+        App_SharedErrorTable_HasAnyCriticalErrorSet(error_table);
+    bool bms_positive_air_closed =
+        App_CanRx_BMS_AIR_STATES_GetSignal_AIR_POSITIVE(can_rx_interface) == 1;
+    bool bms_negative_air_closed =
+        App_CanRx_BMS_AIR_STATES_GetSignal_AIR_NEGATIVE(can_rx_interface) == 1;
+    uint8_t start_switch_position =
+        App_CanRx_DIM_SWITCHES_GetSignal_START_SWITCH(can_rx_interface);
+    uint8_t break_actuated =
+        App_CanRx_FSM_BRAKE_GetSignal_BRAKE_IS_ACTUATED(can_rx_interface);
+
+    App_StartSwitch_SetSwitchToggled(start_switch, start_switch_position);
```

Comment:
It would be better to store the current start switch position inside the start struct and update that value by calling SetSwitchPosition here instead.